test/widgets/recent_dm_conversations_test.dart

```diff
@@ -148,8 +145,38 @@ void main() {
         }
       }
 
+      Future<void> markMessageAsRead(WidgetTester tester, Message message) async {
+        final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+        store.handleEvent(
+          UpdateMessageFlagsAddEvent(id: message.id, flag: MessageFlag.read, all: false, messages: [message.id]),
+        );
+        await tester.pump();
+      }
+
+      void checkUnreadCount(WidgetTester tester, int expectedCount) {
+        // Should appear and be laid out even when expectedCount is zero; read on.
+        final textWidget = tester.widget<Text>(find.descendant(
+          of: find.byType(RecentDmConversationsItem),
+          matching: find.textContaining(RegExp(r'^\d+$'),
+        )));
+
+        final visibilityWidget = tester.widget<Visibility>(
+          find.ancestor(of: find.byWidget(textWidget), matching: find.byType(Visibility)),
+        );
+
+        // When the count is zero, don't show an unread-count widget. Do keep a
+        // "0" placeholder, to approximately match the width of a nonzero
+        // one-digit unread count, given font-size settings etc. That's to
+        // reduce layout jumping of the title text (which wraps and is
+        // truncated) in the common case where a DM conversation goes between
+        // zero and a small number of unreads.
```

Comment:
There might be a way to test this without depending so much on the implementation (in particular the use of the `Visibility` widget with specific params). But I don't yet know how that might look.

Reply:
> instead of having the whole unread-count marker widget tree be there when there are no unreads, just reserve some constant width in pixels.

Hmm. If it really is constant, then we can't avoid it being consistently wrong (maybe very wrong) for some users, right? The width should approximate the width of a single-digit unread counter, and that width will change with the user's choice of font size, whether to force bold, etc.

Reply:
If we can access the effective font size (accounting for the user's font-size setting), we can probably use a width with a component that's proportional to that. Ignoring bold (and the user's force-bold setting), I assume the width of a single digit will be proportional to the effective font size, and maybe we can find the ratio experimentally? It helps that we're providing the font ("Source Sans 3"); I assume the user can't clobber that with their own choice of font but I actually don't know for sure.

Reply:
Well, I think most of all it comes down for me to the fact that it's fine if things sometimes jump slightly here (because it'll be rare for it to do so while the user is looking), and that reserving a constant width is much simpler than any solution involving a placeholder widget.

The placeholder widget doesn't completely eliminate jumping anyway, because the width varies between different values of the counter.

If the width of a counter is increased for a given user because they have a larger font size, that's actually a situation where it's probably best if we *don't* reserve space for the counter, and failing that we should at least not increase the space we reserve. That's because the increased font size means that space is at a premium for the text that actually is on the screen and they want to read. There's some discussion in a slightly different context here:
https://github.com/zulip/zulip-mobile/blob/main/docs/background/webview.md#why-mostly-px-and-not-rem

Reply:
> If the width of a counter is increased for a given user because they have a larger font size, that's actually a situation where it's probably best if we *don't* reserve space for the counter, and failing that we should at least not increase the space we reserve. That's because the increased font size means that space is at a premium for the text that actually is on the screen and they want to read. There's some discussion in a slightly different context here:
https://github.com/zulip/zulip-mobile/blob/main/docs/background/webview.md#why-mostly-px-and-not-rem

Oh, good point. Also: one thing we haven't considered is just not reserving empty space for the counter in any case. That would depart from Vlad's [design in Figma](https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?type=design&node-id=442-18279&mode=design&t=OmJk3hGS1DzlYhaS-0) (so tagging @terpimost), but as you said:

> it's fine if things sometimes jump slightly here (because it'll be rare for it to do so while the user is looking)

and it is nice to not cut off text that we have room to show.